### PR TITLE
Prepare golang-github-* packages for release package requirements

### PR DESCRIPTION
**Pull request by olavphibm**
The packages golang-github-russross-blackfriday and
golang-github-shurcooL-sanitized_anchor_name need to provide a main
package and also a release bump to work with our idea of a Host OS release
package.

### CODE REVIEW

File: golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec [hunk 1]
```diff
@@ -29,6 +29,9 @@ BuildArch:  noarch
 ExclusiveArch:  %{ix86} x86_64 %{arm}
```
Comment (alanoe) [inline golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec]: Commit longer msg is not clear enough, I had to open the rpm specs to understand. I guess you meant:

Those packages preivously only had "*-devel" RPMs => Those rpm packages spec files previously only installed "*-devel" subpackages

?

- I see a %install main block (executed by the main package) which installs several .go files. The "only had" in the commit msg makes me think that the main package did not install anything.
- we have defined today that release rpm won't install all rpms anymore, so commit longer msg must be updated to mention the metadata rpm which will install everything
- Regarding metadata RPMs and package groups, I suggest we check if Linux distros usually have *-devel packages listed in their packages groups and then we decide if we want to have *-devel packages in OP Host OS packages groups which will contain golang-* packages.

Comment (olavphibm) [inline golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec]: The spec files do not install RPMs, they only define how to build them.
- The main package did not even exist. It did not have a `%files` section, so it was not built. `%install` executes when compiling the RPM, not when installing in the target system. See the related section at https://fedoraproject.org/wiki/How_to_create_an_RPM_package
- OK
- OK, this is not related to this PR though.

File: golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 Name:       golang-%{provider}-%{project}-%{repo}
 Version:    1.2
-Release:    5s%{gitcommittag}%{?dist}
+Release:    6%{gitcommittag}%{?dist}
```
Comment (olavphibm) [inline golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec]: For consistency with other packages, like said in the commit message.

File: golang-github-shurcooL-sanitized_anchor_name/CentOS/7/golang-github-shurcooL-sanitized_anchor_name.spec [hunk 3]
```diff
@@ -10,12 +10,13 @@
...
 %global commit          1dba4b3954bc059efc3991ec364f9f9a35f597d2
 %global shortcommit     %(c=%{commit}; echo ${c:0:7})
+%global gitcommittag    .git%{shortcommit}
 %global gccgo_version  >= 5
 %global golang_version >= 1.2.1-3
 
 Name:           golang-%{provider}-%{project}-%{repo}
 Version:        0
-Release:        0.1.sgit%{shortcommit}%{?dist}
+Release:        1%{gitcommittag}%{?dist}
```
Comment (olavphibm) [inline golang-github-shurcooL-sanitized_anchor_name/CentOS/7/golang-github-shurcooL-sanitized_anchor_name.spec]: For consistency with other packages, like said in the commit message.

File: golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec (hunk 1 above)
Comment (alanoe) [inline golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec]: Thanks for clarifying

File: golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec (hunk 2 above)
Comment (alanoe) [inline golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec]: I'd prefer to alter the rpm specs as less as possible, this will make updating to newer rpm specs easier in the future. But up to you.

Comment (olavphibm) [inline golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec]: Currently, a lot of packages have big changes in the spec files, so this change its not a big issue Furthermore, the release field has no meaning across distros, and this was gotten from Fedora.

Comment (alanoe) [inline golang-github-russross-blackfriday/CentOS/7/golang-github-russross-blackfriday.spec]: Agreed
